### PR TITLE
[#2513] Refactor generic ZGW client factories to concrete equivalents

### PR DESCRIPTION
**Pull request by swrichards**
The generic ZGW client factory has the drawback of both typing ambiguity (generic `APIClient` versus concrete `ZakenClient`), but also makes it harder to do code analysis for the purposes of migrating to a multi-ZGW-backend approach. This refactor will address both concerns.

[Taiga #2513](https://taiga.maykinmedia.nl/project/open-inwoner/us/2485)

### CODE REVIEW

**Review by pi-sigma — APPROVED** [review]
Nice! Please create an issue for refactoring the `openklant` clients as well to make it uniform.

File: src/open_inwoner/configurations/bootstrap/zgw.py
Comment (Bartvaderkin) [inline src/open_inwoner/configurations/bootstrap/zgw.py:372]: Note the `build_form_client` is the only one that isn't named plural.

**Review by Bartvaderkin — APPROVED** [review]

**Review by Bartvaderkin — APPROVED** [review]

**Comment by alextreme** [comment]
Can be merged after Paul approves

**Review by pi-sigma — APPROVED** [review]

**Pull request merged**
